File: Rio/Rio.Web/Modules/ServerTypes/Workspace/StudentForm.ts
﻿import { StringEditor, EmailEditor, DateEditor, EnumEditor, TextAreaEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface StudentForm {
    StudentId: StringEditor;
    RollNo: StringEditor;
    FullName: StringEditor;
    Email: EmailEditor;
    Mobile: StringEditor;
    Dob: DateEditor;
    Gender: EnumEditor;
    Note: TextAreaEditor;
    Comments: TextAreaEditor;
}

export class StudentForm extends PrefixedContext {
    static formKey = 'Workspace.Student';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!StudentForm.init)  {
            StudentForm.init = true;

            var w0 = StringEditor;
            var w1 = EmailEditor;
            var w2 = DateEditor;
            var w3 = EnumEditor;
            var w4 = TextAreaEditor;

            initFormType(StudentForm, [
                'StudentId', w0,
                'RollNo', w0,
                'FullName', w0,
                'Email', w1,
                'Mobile', w0,
                'Dob', w2,
                'Gender', w3,
                'Note', w4,
                'Comments', w4
            ]);
        }
    }
}

File: Rio/Rio.Web/Modules/Workspace/Student/StudentForm.cs
using Rio.Web.Enums;
using Serenity.ComponentModel;
using Serenity.Web;
using System;

namespace Rio.Workspace.Forms
{
    [FormScript("Workspace.Student")]
    [BasedOnRow(typeof(StudentRow), CheckNames = true)]
    public class StudentForm
    {
        [HalfWidth]
        public Guid StudentId { get; set; }
        [HalfWidth]
        public long RollNo { get; set; }
        [HalfWidth]
        public string FullName { get; set; }
        [EmailEditor]
        public string Email { get; set; }
        [HalfWidth]
        public string Mobile { get; set; }
        [HalfWidth]
        public DateTime Dob { get; set; }
        [HalfWidth]
        public Gender Gender { get; set; }
        [TextAreaEditor(Rows = 3)]
        public string Note { get; set; }
        [TextAreaEditor(Rows = 3)]
        public string Comments { get; set; }
    }
}